### DISCUSSION TITLE
ergm levels

### DISCUSSION THREAD

**Discussion by giladravid** [statnet/ergm#555] [Q&A]
I need to build a model where I can control the baseline and the exclude list of levels in nodeifactor term. For example, I want to specify that the baseline will be the X level and exclude levels Y and Z due to empty examples in the network (to avoid the Inf coefficient).

**Comment by krivit** [comment]
It depends on *why* levels Y and Z have no nonzeros:

- If they correspond to structural zeros, than a constraint or an offset should be used to fix the corresponding dyads. Then, levels Y and Z could be excluded along with X, but they wouldn't have any effect on the parameter estimates.
- If they are not structural zeros, then -Inf is the best estimator available, though if it is because there are few nodes in Y and Z, it may be worthwhile to merge some number of smallest categories. (Nodal attribute interface has some useful shortcuts for this.)

**Reply by mbojan** [reply]
fyi @giladravid , the nodal attributes interface is https://www.rdocumentation.org/packages/ergm/versions/4.5.0/topics/nodal_attributes or `?nodal_attributes`.

**Reply by giladravid** [reply]
@krivit , @mbojan, many thanks.
I think the generalized question is how we can determine which level among the excluded list will be the baseline. 
In this particular case, the edges are structural zeros. How do we implement constraints or exclude them?

**Reply by krivit** [reply]
All of the excluded levels are pooled together to create the baseline. However, if the zeros are structural, fixing them will exclude them from being modelled (i.e., from both the "numerator" and the "denominator"), and so they won't affect the estimator.

I think something along the lines of `constraints = ~egocentric(~a %in% c("Y", "Z"), "in")`, where `a` is the nodal attribute name, should do the trick. (See `ergmConstraint?egocentric` and/or the 2023 ergm 4 package paper for help.)
